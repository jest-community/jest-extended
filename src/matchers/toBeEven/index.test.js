import each from 'jest-each';

import matcher from './';

expect.extend(matcher);

describe('.toBeEven', () => {
  it('passes when given even number', () => {
    expect(2).toBeEven();
  });

  each([
    [false],
    [true],
    [''],
    [1],
    [{}],
    [() => {}],
    [undefined],
    [null],
    [NaN]
  ]).it('fails when not given an even number', given => {
    expect(() => expect(given).toBeEven()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeEven', () => {
  it('fails when given an even number', () => {
    expect(() => expect(2).not.toBeEven()).toThrowErrorMatchingSnapshot();
  });

  each([
    [false],
    [true],
    [''],
    [1],
    [[]],
    [{}],
    [() => {}],
    [undefined],
    [null],
    [NaN]
  ]).it('passes when not given an even number: %s', given => {
    expect(given).not.toBeEven();
  });
});
