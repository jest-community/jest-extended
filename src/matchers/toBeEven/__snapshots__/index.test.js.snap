// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeEven fails when given an even number 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeEven(</><dim>)</>

Expected value to not be an even number received:
 <red>2</>"
`;

exports[`.toBeEven fails when not given an even number 1`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>false</>"
`;

exports[`.toBeEven fails when not given an even number 2`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>true</>"
`;

exports[`.toBeEven fails when not given an even number 3`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>\\"\\"</>"
`;

exports[`.toBeEven fails when not given an even number 4`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>1</>"
`;

exports[`.toBeEven fails when not given an even number 5`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>{}</>"
`;

exports[`.toBeEven fails when not given an even number 6`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>[Function anonymous]</>"
`;

exports[`.toBeEven fails when not given an even number 7`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>undefined</>"
`;

exports[`.toBeEven fails when not given an even number 8`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>null</>"
`;

exports[`.toBeEven fails when not given an even number 9`] = `
"<dim>expect(</><red>received</><dim>).toBeEven(</><dim>)</>

Expected value to be an even number received:
 <red>NaN</>"
`;
